ui/packages/shared/profile/src/ProfileView.tsx

```diff
@@ -53,29 +83,26 @@ export const ProfileView = ({
     setCurPath([]);
   }, [profileSource]);
 
-  useEffect(() => {
-    let showLoaderTimeout;
-    if (isLoading && !isLoaderVisible) {
-      // if the request takes longer than half a second, show the loading icon
-      showLoaderTimeout = setTimeout(() => {
-        setIsLoaderVisible(true);
-      }, 500);
-    } else {
-      setIsLoaderVisible(false);
-    }
-    return () => clearTimeout(showLoaderTimeout);
-  }, [isLoading]);
+  const isLoaderVisible = useDelayedLoader(flamegraphData?.loading);
```

Comment:
Shouldn't this loader be based on the data that is need to render the particular current view? For example, if currentView is 'table' we should show a loader when the table data is loading rather than the flamegraph data.

Reply:
Great catch, fixed it.